terraform/errors.go

```diff
@@ -7,9 +7,19 @@ type CustomError struct {
 }
 
 func (err *CustomError) Error() string {
+	// TODO: include more details here with user information
 	return err.message
 }
 
+func isUserError(err error) bool {
```

Comment:
I'm using this instead of returning `*CustomError` from the functions and accessing the field because apparently Golang does not treat `nil` as `nil` when the return type is an interface: https://go.dev/doc/faq#nil_error